Complete bottles_of_beer assignment

RyanSpittler opened the pull request:
[#88836366]

RyanSpittler committed:
Complete bottles_of_beer assignment

[#88836366]

jasonnoble commented on methods_and_recursion/bottles_of_beer.rb:
```diff
@@ -0,0 +1,29 @@
+       " #{number1} bottle#{'s' unless number1 == 1} of beer."
+  puts "Take one down and pass it around,"\
+       " #{number2} bottle#{'s' unless number2 == 1} of beer on the wall."
+  puts
```
Best practice is to use `puts ''`.  A `puts` by itself makes a future developer wonder if you intended to not having anything after the puts.  Using `puts ''` makes it more explicit.

jasonnoble commented on methods_and_recursion/bottles_of_beer.rb:
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+
+def sing_a_verse(number1, number2)
+  puts "#{number1} bottle#{'s' unless number1 == 1} of beer on the wall,"\
```
Could you dry up your code by pulling out a `bottles_text` variable?

jasonnoble commented on methods_and_recursion/bottles_of_beer.rb:
```diff
@@ -0,0 +1,29 @@
+  sing_a_song(remaining - 1) if remaining > 0
+end
+
+sing_a_song(99)
```
:heart:

RyanSpittler committed:
Dry up code

- Factored out the bottles line into a method that can have the wall or not.

[#88836366]
